examples/tutorials/date_time_charts.py

```diff
@@ -0,0 +1,340 @@
+    color="yellow",
+)
+
+fig.show()
+
+########################################################################################
+# Using :meth:`pandas.date_range`
+# ------------------------------------------------
+#
+# In the following example, a ``pandas.date_range`` object is used to pass date
+# data to the PyGMT figure. This object is set using the pandas method ``date_range()``.
+# This particular object contains 7 different ``pandas.Timestamp`` objects, with the
+# number being manipulated by the periods argument. Each period begins at the start
+# of a business quarter as denoted by BQS when passed to the freq argument. The inital
+# date is first argument that is passed to ``date_range()`` and it marks the first
+# data in the object ``x`` that will be plotted.
+
+x = pd.date_range("2018-03-01", periods=7, freq="BQS")
+y = [4, 5, 6, 8, 6, 3, 5]
+
+fig = pygmt.Figure()
+fig.plot(
+    projection="X10c/10c",
+    region=[datetime.datetime(2017, 12, 31), datetime.datetime(2019, 12, 31), 0, 10],
+    frame=["WSen", "ag"],
+    x=x,
+    y=y,
+    style="i0.4c",
+    pen="1p",
+    color="purple",
+)
+
+fig.show()
+
+########################################################################################
+# Using :meth:`xarray.DataArray`
+# -------------------------------------
+#
+# In this example, instead of using a ``pd.date_range`` object, ``x`` is initialized
+# as an ``xarray.DataArray`` object. Such object provide a wrapper around traditional
+# data formats allowing this data to have varrying labeled dimensions and support
+# operations that use various pieces of metadata. The following code uses a
+# ``pandas.date_range`` object to fill the DataArray with data, yet this is not
+# essential for the creation of a valid DataArray.
+
+x = xr.DataArray(data=pd.date_range(start="2020-01-01", periods=4, freq="Q"))
+y = [4, 7, 5, 6]
+
+fig = pygmt.Figure()
+fig.plot(
+    projection="X10c/10c",
+    region=[datetime.datetime(2020, 1, 1), datetime.datetime(2021, 4, 1), 0, 10],
+    frame=["WSen", "ag"],
+    x=x,
+    y=y,
+    style="n0.4c",
+    pen="1p",
+    color="red",
+)
+fig.show()
+
+# Code
+
+########################################################################################
+# Generating Region Using :meth:`pygmt.info`
+# ----------------------
+#
+# Explanation of supported parameters + bug at #597.
+
+data = [
+    ["20200712", 1000],
+    ["20200714", 1235],
+    ["20200716", 1336],
+    ["20200719", 1176],
+    ["20200721", 1573],
+    ["20200724", 1893],
+    ["20200729", 1634],
+]
+
+df = pd.DataFrame(data, columns=["Date", "Score"])
+
+df["Date"] = pd.to_datetime(df["Date"], format="%Y%m%d")
+
+fig = pygmt.Figure()
+region = pygmt.info(
+    table=df[["Date", "Score"]],
+    per_column=True,
+    spacing=(5000, 1200),
+)
+
+fig.plot(
+    region=region,
+    projection="X15c/10c",
+    frame=["WSen", "afg"],
+    x=df.Date,
+    y=df.Score,
+    style="c0.4c",
+    pen="1p",
+    color="green3",
+)
+fig.show()
```

Comment:
I'm also having some issues with this example where the date is not actually showing on the chart and I'm not sure what could be going wrong.
![Screenshot from 2021-04-16 00-49-06](https://user-images.githubusercontent.com/55197145/114973029-95edeb80-9e4d-11eb-8a31-30c5f3f3641c.png)

Reply:
I have since fixed this issue but I'm still having difficulties in figuring out how the spacing parameter works for datetime data

Reply:
It's also unclear to me how `pygmt.info()` deal with datetimes when `spacing` parameter is used. Internally, any datetime is converted to seconds since the epoch (1970-01-01 midnight). In your example, I think `5000` means 5000 seconds.

I expect a spacing like `1d` (1 day) should work, but it doesn't. Perhaps GMT doesn't support it. I need more tests.